elasticgraph-warehouse/spec/unit/elastic_graph/warehouse/schema_definition/results_extension_spec.rb

```diff
@@ -0,0 +1,167 @@
+# Copyright 2024 - 2025 Block, Inc.
+#
+# Use of this source code is governed by an MIT-style
+# license that can be found in the LICENSE file or at
+# https://opensource.org/licenses/MIT.
+#
+# frozen_string_literal: true
+
+require "elastic_graph/warehouse/schema_definition/api_extension"
+
+module ElasticGraph
+  module Warehouse
+    module SchemaDefinition
+      RSpec.describe ResultsExtension, :warehouse_schema do
+        it "generates warehouse config for types with warehouse_table definitions" do
+          results = define_warehouse_schema do |s|
+            s.object_type "Product" do |t|
+              t.field "id", "ID"
+              t.field "name", "String"
+              t.field "price", "Float"
+              t.warehouse_table "products"
+            end
+
+            s.object_type "Order" do |t|
+              t.field "order_id", "ID"
+              t.field "total", "Float"
+              t.warehouse_table "orders"
+            end
+          end
+
+          config = results.warehouse_config
+
+          expect(config).to be_a(Hash)
+          expect(config.keys).to contain_exactly("orders", "products")
+          expect(config["products"]["table_schema"]).to include("CREATE TABLE IF NOT EXISTS products")
+          expect(config["orders"]["table_schema"]).to include("CREATE TABLE IF NOT EXISTS orders")
```

Comment:
```suggestion
          expect(config).to eq({
            "orders" => {
              "table_schema" => <<~EOS.strip
                CREATE TABLE IF NOT EXISTS orders (
                  order_id STRING,
                  total DOUBLE
                )
              EOS
            },
            "products" => {
              "table_schema" => <<~EOS.strip
                CREATE TABLE IF NOT EXISTS products (
                  id STRING,
                  name STRING,
                  price DOUBLE
                )
              EOS
            }
          })
```

I think it's a bit more robust to assert on the entire thing.